### ISSUE TITLE
postgresql-k8s/0 awaiting for cluster to start infinity loop

### ISSUE THREAD

**Issue by mcfly722** [canonical/postgresql-k8s-operator#473]
## Steps to reproduce
```juju deploy postgresql-k8s --channel 14/stable --trust```

## Expected behavior
PostgreSQL unit in active state as Primary

## Actual behavior
postgresql-k8s/0 unit in **awaiting for cluster to start** state:
```
Model            Controller          Cloud/Region        Version  SLA          Timestamp
postgresql-maas  microk8s-localhost  microk8s/localhost  3.5.0    unsupported  19:05:37Z

App             Version  Status   Scale  Charm           Channel    Rev  Address         Exposed  Message
postgresql-k8s  14.10    waiting      1  postgresql-k8s  14/stable  193  10.152.183.192  no       installing agent

Unit               Workload  Agent  Address      Ports  Message
postgresql-k8s/0*  waiting   idle   10.1.251.89         awaiting for cluster to start
```

Deleting and recreating model with storage not resolves issue. Same behaviour occurs.

If you try to obtain PG credentils:
```
juju run postgresql-k8s/leader get-password
Running operation 1 with 1 task
  - task 2 on unit-postgresql-k8s-0

Waiting for task 2...
ERROR timed out waiting for results from: unit postgresql-k8s/0
```

Analysing logs looks like some process enters to infinity loop and waiting for something...

## Versions
Operating system: **Ubuntu 22.04.4 LTS**
Juju CLI: **3.5.0-genericlinux-and64**
Juju agent: **3.5.0**
Charm revision: **193**
microk8s: **MicroK8s v1.28.7 revision 6532**

## Log output
```
juju debug-log --include postgresql-k8s/0

unit-postgresql-k8s-0: 19:00:19 INFO unit.postgresql-k8s/0.juju-log Kubernetes service 'postgresql-k8s' patched successfully
unit-postgresql-k8s-0: 19:00:20 INFO juju.worker.uniter.operation ran "update-status" hook (via hook dispatching script: dispatch)
unit-postgresql-k8s-0: 19:03:34 INFO unit.postgresql-k8s/0.juju-log Updating Patroni config file
unit-postgresql-k8s-0: 19:03:34 INFO unit.postgresql-k8s/0.juju-log Updated health checks
unit-postgresql-k8s-0: 19:04:35 INFO unit.postgresql-k8s/0.juju-log Kubernetes service 'postgresql-k8s' patched successfully
unit-postgresql-k8s-0: 19:04:36 INFO juju.worker.uniter.operation ran "update-status" hook (via hook dispatching script: dispatch)
unit-postgresql-k8s-0: 19:07:38 INFO unit.postgresql-k8s/0.juju-log Updating Patroni config file
unit-postgresql-k8s-0: 19:07:38 INFO unit.postgresql-k8s/0.juju-log Updated health checks
unit-postgresql-k8s-0: 19:08:39 INFO unit.postgresql-k8s/0.juju-log Kubernetes service 'postgresql-k8s' patched successfully
unit-postgresql-k8s-0: 19:08:40 INFO juju.worker.uniter.operation ran "update-status" hook (via hook dispatching script: dispatch)
unit-postgresql-k8s-0: 19:12:38 INFO unit.postgresql-k8s/0.juju-log Updating Patroni config file
unit-postgresql-k8s-0: 19:12:39 INFO unit.postgresql-k8s/0.juju-log Updated health checks
unit-postgresql-k8s-0: 19:13:39 INFO unit.postgresql-k8s/0.juju-log Kubernetes service 'postgresql-k8s' patched successfully
unit-postgresql-k8s-0: 19:13:40 INFO juju.worker.uniter.operation ran "update-status" hook (via hook dispatching script: dispatch)
unit-postgresql-k8s-0: 19:14:49 INFO unit.postgresql-k8s/0.juju-log Updating Patroni config file
unit-postgresql-k8s-0: 19:14:50 INFO unit.postgresql-k8s/0.juju-log Updated health checks
unit-postgresql-k8s-0: 19:18:36 INFO unit.postgresql-k8s/0.juju-log Updating Patroni config file
unit-postgresql-k8s-0: 19:19:37 INFO unit.postgresql-k8s/0.juju-log Updated health checks
unit-postgresql-k8s-0: 19:20:38 INFO unit.postgresql-k8s/0.juju-log Kubernetes service 'postgresql-k8s' patched successfully
unit-postgresql-k8s-0: 19:20:39 INFO juju.worker.uniter.operation ran "update-status" hook (via hook dispatching script: dispatch)
.... and so on
```

## Additional context
```
microk8s.kubectl get all -A

NAMESPACE                       NAME                                         READY   STATUS    RESTARTS      AGE
controller-microk8s-localhost   pod/modeloperator-bbb68d78b-knfmt            1/1     Running   2 (89m ago)   7h59m
network                         pod/haproxy-sxv8t                            1/1     Running   2 (89m ago)   8h
kube-system                     pod/coredns-864597b5fd-xsvdl                 1/1     Running   2 (89m ago)   8h
metallb-system                  pod/speaker-sgdxv                            1/1     Running   2 (89m ago)   8h
kube-system                     pod/calico-node-2pwmx                        1/1     Running   2 (89m ago)   8h
kube-system                     pod/calico-kube-controllers-77bd7c5b-fxkdx   1/1     Running   2 (89m ago)   8h
kube-system                     pod/calico-node-4v4xr                        1/1     Running   1 (87m ago)   8h
metallb-system                  pod/speaker-xswhh                            1/1     Running   1 (87m ago)   8h
kube-system                     pod/hostpath-provisioner-7df77bc496-9gfgs    1/1     Running   1 (87m ago)   8h
network                         pod/haproxy-dgmsx                            1/1     Running   1 (87m ago)   8h
controller-microk8s-localhost   pod/controller-0                             3/3     Running   3 (87m ago)   8h
kube-system                     pod/calico-node-525jv                        1/1     Running   2 (85m ago)   8h
metallb-system                  pod/speaker-4kqcw                            1/1     Running   2 (85m ago)   8h
default                         pod/echoserver-6545b468b-grqfr               1/1     Running   2 (85m ago)   8h
network                         pod/haproxy-lkl96                            1/1     Running   2 (85m ago)   8h
metallb-system                  pod/controller-5f7bb57799-z28gd              1/1     Running   2 (85m ago)   8h
postgresql-maas                 pod/modeloperator-6dd67b8594-pxz6t           1/1     Running   0             75m
postgresql-maas                 pod/postgresql-k8s-0                         1/2     Running   0             74m

NAMESPACE                       NAME                               TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                  AGE
default                         service/kubernetes                 ClusterIP   10.152.183.1     <none>        443/TCP                  8h
kube-system                     service/kube-dns                   ClusterIP   10.152.183.10    <none>        53/UDP,53/TCP,9153/TCP   8h
metallb-system                  service/webhook-service            ClusterIP   10.152.183.31    <none>        443/TCP                  8h
network                         service/proxy                      ClusterIP   10.152.183.179   <none>        3126/TCP                 8h
controller-microk8s-localhost   service/controller-service         ClusterIP   10.152.183.209   <none>        17070/TCP                8h
controller-microk8s-localhost   service/modeloperator              ClusterIP   10.152.183.136   <none>        17071/TCP                7h59m
postgresql-maas                 service/modeloperator              ClusterIP   10.152.183.153   <none>        17071/TCP                75m
postgresql-maas                 service/postgresql-k8s-endpoints   ClusterIP   None             <none>        <none>                   74m
postgresql-maas                 service/postgresql-k8s             ClusterIP   10.152.183.192   <none>        5432/TCP,8008/TCP        74m
postgresql-maas                 service/postgresql-k8s-primary     ClusterIP   10.152.183.131   <none>        8008/TCP,5432/TCP        73m
postgresql-maas                 service/postgresql-k8s-replicas    ClusterIP   10.152.183.159   <none>        8008/TCP,5432/TCP        73m

NAMESPACE        NAME                         DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE
kube-system      daemonset.apps/calico-node   3         3         3       3            3           kubernetes.io/os=linux   8h
metallb-system   daemonset.apps/speaker       3         3         3       3            3           kubernetes.io/os=linux   8h
network          daemonset.apps/haproxy       3         3         3       3            3           <none>                   8h

NAMESPACE                       NAME                                      READY   UP-TO-DATE   AVAILABLE   AGE
kube-system                     deployment.apps/hostpath-provisioner      1/1     1            1           8h
kube-system                     deployment.apps/calico-kube-controllers   1/1     1            1           8h
kube-system                     deployment.apps/coredns                   1/1     1            1           8h
controller-microk8s-localhost   deployment.apps/modeloperator             1/1     1            1           7h59m
default                         deployment.apps/echoserver                1/1     1            1           8h
metallb-system                  deployment.apps/controller                1/1     1            1           8h
postgresql-maas                 deployment.apps/modeloperator             1/1     1            1           75m

NAMESPACE                       NAME                                               DESIRED   CURRENT   READY   AGE
kube-system                     replicaset.apps/hostpath-provisioner-7df77bc496    1         1         1       8h
controller-microk8s-localhost   replicaset.apps/modeloperator-bbb68d78b            1         1         1       7h59m
kube-system                     replicaset.apps/coredns-864597b5fd                 1         1         1       8h
kube-system                     replicaset.apps/calico-kube-controllers-77bd7c5b   1         1         1       8h
default                         replicaset.apps/echoserver-6545b468b               1         1         1       8h
metallb-system                  replicaset.apps/controller-5f7bb57799              1         1         1       8h
postgresql-maas                 replicaset.apps/modeloperator-6dd67b8594           1         1         1       75m

NAMESPACE                       NAME                              READY   AGE
controller-microk8s-localhost   statefulset.apps/controller       1/1     8h
postgresql-maas                 statefulset.apps/postgresql-k8s   0/1     74m
```

```
microk8s.kubectl logs postgresql-k8s-0 -n postgresql-maas

Defaulted container "charm" out of: charm, postgresql, charm-init (init)
2024-05-08T18:08:26.366Z [pebble] HTTP API server listening on ":38812".
2024-05-08T18:08:26.366Z [pebble] Started daemon.
2024-05-08T18:08:26.376Z [pebble] POST /v1/services 5.380221ms 202
2024-05-08T18:08:26.380Z [pebble] Service "container-agent" starting: /charm/bin/containeragent unit --data-dir /var/lib/juju --append-env "PATH=$PATH:/charm/bin" --show-log --charm-modified-version 0
2024-05-08T18:08:26.458Z [container-agent] 2024-05-08 18:08:26 INFO juju.cmd supercommand.go:56 running containerAgent [3.5.0 01a596920ff5a632babf5aacf485cc78fe1ed5df gc go1.21.9]
2024-05-08T18:08:26.458Z [container-agent] starting containeragent unit command
2024-05-08T18:08:26.458Z [container-agent] containeragent unit "unit-postgresql-k8s-0" start (3.5.0 [gc])
2024-05-08T18:08:26.459Z [container-agent] 2024-05-08 18:08:26 INFO juju.cmd.containeragent.unit runner.go:592 start "unit"
2024-05-08T18:08:26.459Z [container-agent] 2024-05-08 18:08:26 INFO juju.worker.upgradesteps worker.go:60 upgrade steps for 3.5.0 have already been run.
2024-05-08T18:08:26.464Z [container-agent] 2024-05-08 18:08:26 INFO juju.worker.probehttpserver server.go:157 starting http server on 127.0.0.1:65301
2024-05-08T18:08:26.504Z [container-agent] 2024-05-08 18:08:26 INFO juju.api apiclient.go:996 cannot resolve "controller-service.controller-microk8s-localhost.svc.cluster.local": lookup controller-service.controller-microk8s-localhost.svc.cluster.local: operation was canceled
2024-05-08T18:08:26.505Z [container-agent] 2024-05-08 18:08:26 INFO juju.api apiclient.go:629 connection established to "wss://10.152.183.209:17070/model/0d044226-7a1b-4a6a-8bf6-3125d927551b/api"
2024-05-08T18:08:26.517Z [container-agent] 2024-05-08 18:08:26 INFO juju.worker.apicaller connect.go:163 [0d0442] "unit-postgresql-k8s-0" successfully connected to "10.152.183.209:17070"
2024-05-08T18:08:26.566Z [container-agent] 2024-05-08 18:08:26 INFO juju.api apiclient.go:996 cannot resolve "controller-service.controller-microk8s-localhost.svc.cluster.local": lookup controller-service.controller-microk8s-localhost.svc.cluster.local: operation was canceled
2024-05-08T18:08:26.566Z [container-agent] 2024-05-08 18:08:26 INFO juju.api apiclient.go:629 connection established to "wss://10.152.183.209:17070/model/0d044226-7a1b-4a6a-8bf6-3125d927551b/api"
2024-05-08T18:08:26.576Z [container-agent] 2024-05-08 18:08:26 INFO juju.worker.apicaller connect.go:163 [0d0442] "unit-postgresql-k8s-0" successfully connected to "10.152.183.209:17070"
2024-05-08T18:08:26.601Z [container-agent] 2024-05-08 18:08:26 INFO juju.worker.migrationminion worker.go:142 migration phase is now: NONE
2024-05-08T18:08:26.601Z [container-agent] 2024-05-08 18:08:26 INFO juju.worker.logger logger.go:120 logger worker started
2024-05-08T18:08:26.613Z [container-agent] 2024-05-08 18:08:26 WARNING juju.worker.proxyupdater proxyupdater.go:241 unable to set snap core settings [proxy.http= proxy.https= proxy.store=]: exec: "snap": executable file not found in $PATH, output: ""
2024-05-08T18:08:26.621Z [container-agent] 2024-05-08 18:08:26 INFO juju.worker.leadership tracker.go:194 postgresql-k8s/0 promoted to leadership of postgresql-k8s
2024-05-08T18:08:26.635Z [container-agent] 2024-05-08 18:08:26 INFO juju.worker.caasupgrader upgrader.go:113 abort check blocked until version event received
2024-05-08T18:08:26.635Z [container-agent] 2024-05-08 18:08:26 INFO juju.worker.caasupgrader upgrader.go:119 unblocking abort check
2024-05-08T18:08:26.657Z [container-agent] 2024-05-08 18:08:26 INFO juju.agent.tools symlinks.go:20 ensure jujuc symlinks in /var/lib/juju/tools/unit-postgresql-k8s-0
2024-05-08T18:08:26.713Z [container-agent] 2024-05-08 18:08:26 INFO juju.worker.uniter uniter.go:363 unit "postgresql-k8s/0" started
2024-05-08T18:08:26.727Z [container-agent] 2024-05-08 18:08:26 INFO juju.worker.uniter uniter.go:689 resuming charm install
2024-05-08T18:08:26.734Z [container-agent] 2024-05-08 18:08:26 INFO juju.worker.uniter.charm bundles.go:82 downloading ch:amd64/jammy/postgresql-k8s-193 from API server
2024-05-08T18:08:26.735Z [container-agent] 2024-05-08 18:08:26 INFO juju.downloader download.go:112 downloading from ch:amd64/jammy/postgresql-k8s-193
2024-05-08T18:08:27.387Z [pebble] GET /v1/changes/1/wait 1.008737552s 200
2024-05-08T18:08:27.387Z [pebble] Started default services with change 1.
2024-05-08T18:08:28.042Z [container-agent] 2024-05-08 18:08:28 INFO juju.downloader download.go:95 download complete ("ch:amd64/jammy/postgresql-k8s-193")
2024-05-08T18:08:28.488Z [container-agent] 2024-05-08 18:08:28 INFO juju.downloader download.go:175 download verified ("ch:amd64/jammy/postgresql-k8s-193")
2024-05-08T18:08:36.372Z [pebble] Check "readiness" failure 1 (threshold 3): received non-20x status code 418
2024-05-08T18:08:42.572Z [container-agent] 2024-05-08 18:08:42 INFO juju.worker.uniter uniter.go:389 hooks are retried true
2024-05-08T18:08:42.753Z [container-agent] 2024-05-08 18:08:42 INFO juju.worker.uniter resolver.go:185 found queued "install" hook
2024-05-08T18:08:46.366Z [pebble] Check "readiness" failure 2 (threshold 3): received non-20x status code 418
2024-05-08T18:08:46.861Z [container-agent] 2024-05-08 18:08:46 INFO juju-log Running legacy hooks/install.
2024-05-08T18:08:49.084Z [container-agent] 2024-05-08 18:08:49 INFO juju-log Kubernetes service 'postgresql-k8s' patched successfully
2024-05-08T18:08:49.760Z [container-agent] 2024-05-08 18:08:49 INFO juju.worker.uniter.operation runhook.go:186 ran "install" hook (via hook dispatching script: dispatch)
2024-05-08T18:08:52.843Z [container-agent] 2024-05-08 18:08:52 INFO juju.worker.uniter.operation runhook.go:186 ran "upgrade-relation-created" hook (via hook dispatching script: dispatch)
2024-05-08T18:08:55.372Z [container-agent] 2024-05-08 18:08:55 INFO juju.worker.uniter.operation runhook.go:186 ran "database-peers-relation-created" hook (via hook dispatching script: dispatch)
2024-05-08T18:08:56.367Z [pebble] Check "readiness" failure 3 (threshold 3): received non-20x status code 418
2024-05-08T18:08:56.367Z [pebble] Check "readiness" failure threshold 3 hit, triggering action
2024-05-08T18:08:57.421Z [container-agent] 2024-05-08 18:08:57 INFO juju.worker.uniter.operation runhook.go:186 ran "restart-relation-created" hook (via hook dispatching script: dispatch)
2024-05-08T18:08:57.462Z [container-agent] 2024-05-08 18:08:57 INFO juju.worker.uniter resolver.go:185 found queued "leader-elected" hook
2024-05-08T18:09:00.753Z [container-agent] 2024-05-08 18:09:00 INFO juju.worker.uniter.operation runhook.go:186 ran "leader-elected" hook (via hook dispatching script: dispatch)
2024-05-08T18:09:03.565Z [container-agent] 2024-05-08 18:09:03 INFO juju-log Updating Patroni config file
2024-05-08T18:09:03.884Z [container-agent] 2024-05-08 18:09:03 INFO juju-log Added updated layer 'postgresql' to Pebble plan
2024-05-08T18:09:05.967Z [container-agent] 2024-05-08 18:09:05 INFO juju-log Restarted postgresql service
2024-05-08T18:09:05.988Z [container-agent] 2024-05-08 18:09:05 INFO juju-log Updated health checks
2024-05-08T18:09:06.367Z [pebble] Check "readiness" failure 4 (threshold 3): received non-20x status code 418
2024-05-08T18:09:16.367Z [pebble] Check "readiness" failure 5 (threshold 3): received non-20x status code 418
2024-05-08T18:09:26.369Z [pebble] Check "readiness" failure 6 (threshold 3): received non-20x status code 418
2024-05-08T18:09:36.371Z [pebble] Check "readiness" failure 7 (threshold 3): received non-20x status code 418
2024-05-08T18:09:46.368Z [pebble] Check "readiness" failure 8 (threshold 3): received non-20x status code 418
2024-05-08T18:09:56.367Z [pebble] Check "readiness" failure 9 (threshold 3): received non-20x status code 418
2024-05-08T18:10:06.367Z [pebble] Check "readiness" failure 10 (threshold 3): received non-20x status code 418
2024-05-08T18:10:07.203Z [container-agent] 2024-05-08 18:10:07 INFO juju.worker.uniter.operation runhook.go:186 ran "postgresql-pebble-ready" hook (via hook dispatching script: dispatch)
2024-05-08T18:10:10.056Z [container-agent] 2024-05-08 18:10:10 INFO juju-log Updating Patroni config file
2024-05-08T18:10:10.371Z [container-agent] 2024-05-08 18:10:10 INFO juju-log Updated health checks
2024-05-08T18:10:16.366Z [pebble] Check "readiness" failure 11 (threshold 3): received non-20x status code 418
2024-05-08T18:10:26.366Z [pebble] Check "readiness" failure 12 (threshold 3): received non-20x status code 418
2024-05-08T18:10:36.366Z [pebble] Check "readiness" failure 13 (threshold 3): received non-20x status code 418
2024-05-08T18:10:46.368Z [pebble] Check "readiness" failure 14 (threshold 3): received non-20x status code 418
2024-05-08T18:10:56.367Z [pebble] Check "readiness" failure 15 (threshold 3): received non-20x status code 418
2024-05-08T18:11:06.366Z [pebble] Check "readiness" failure 16 (threshold 3): received non-20x status code 418
2024-05-08T18:11:11.518Z [container-agent] 2024-05-08 18:11:11 INFO juju.worker.uniter.operation runhook.go:186 ran "pgdata-storage-attached" hook (via hook dispatching script: dispatch)
2024-05-08T18:11:14.197Z [container-agent] 2024-05-08 18:11:14 INFO juju-log Updating Patroni config file
2024-05-08T18:11:14.538Z [container-agent] 2024-05-08 18:11:14 INFO juju-log Updated health checks
2024-05-08T18:11:16.366Z [pebble] Check "readiness" failure 17 (threshold 3): received non-20x status code 418
2024-05-08T18:11:26.367Z [pebble] Check "readiness" failure 18 (threshold 3): received non-20x status code 418
2024-05-08T18:11:36.368Z [pebble] Check "readiness" failure 19 (threshold 3): received non-20x status code 418
2024-05-08T18:11:46.368Z [pebble] Check "readiness" failure 20 (threshold 3): received non-20x status code 418
2024-05-08T18:11:56.367Z [pebble] Check "readiness" failure 21 (threshold 3): received non-20x status code 418
2024-05-08T18:12:06.366Z [pebble] Check "readiness" failure 22 (threshold 3): received non-20x status code 418
2024-05-08T18:12:15.784Z [container-agent] 2024-05-08 18:12:15 INFO juju.worker.uniter.operation runhook.go:186 ran "config-changed" hook (via hook dispatching script: dispatch)
2024-05-08T18:12:15.806Z [container-agent] 2024-05-08 18:12:15 INFO juju.worker.uniter resolver.go:185 found queued "start" hook
2024-05-08T18:12:16.367Z [pebble] Check "readiness" failure 23 (threshold 3): received non-20x status code 418
2024-05-08T18:12:17.657Z [container-agent] 2024-05-08 18:12:17 INFO juju-log Running legacy hooks/start.
2024-05-08T18:12:20.466Z [container-agent] 2024-05-08 18:12:20 INFO juju-log Updating Patroni config file
2024-05-08T18:12:20.807Z [container-agent] 2024-05-08 18:12:20 INFO juju-log Updated health checks
2024-05-08T18:12:26.368Z [pebble] Check "readiness" failure 24 (threshold 3): received non-20x status code 418
2024-05-08T18:12:36.367Z [pebble] Check "readiness" failure 25 (threshold 3): received non-20x status code 418
2024-05-08T18:12:46.368Z [pebble] Check "readiness" failure 26 (threshold 3): received non-20x status code 418
2024-05-08T18:12:56.369Z [pebble] Check "readiness" failure 27 (threshold 3): received non-20x status code 418
2024-05-08T18:13:06.369Z [pebble] Check "readiness" failure 28 (threshold 3): received non-20x status code 418
2024-05-08T18:13:16.369Z [pebble] Check "readiness" failure 29 (threshold 3): received non-20x status code 418
2024-05-08T18:13:22.545Z [container-agent] 2024-05-08 18:13:22 INFO juju.worker.uniter.operation runhook.go:186 ran "start" hook (via hook dispatching script: dispatch)
2024-05-08T18:13:25.185Z [container-agent] 2024-05-08 18:13:25 INFO juju-log upgrade:0: Updating Patroni config file
2024-05-08T18:13:25.550Z [container-agent] 2024-05-08 18:13:25 INFO juju-log upgrade:0: Updated health checks
2024-05-08T18:15:27.255Z [container-agent] 2024-05-08 18:15:27 INFO juju.worker.uniter.operation runhook.go:186 ran "upgrade-relation-changed" hook (via hook dispatching script: dispatch)
2024-05-08T18:15:29.960Z [container-agent] 2024-05-08 18:15:29 INFO juju-log database-peers:1: Updating Patroni config file
2024-05-08T18:15:30.321Z [container-agent] 2024-05-08 18:15:30 INFO juju-log database-peers:1: Updated health checks
2024-05-08T18:16:31.683Z [container-agent] 2024-05-08 18:16:31 INFO juju.worker.uniter.operation runhook.go:186 ran "database-peers-relation-changed" hook (via hook dispatching script: dispatch)
2024-05-08T18:19:49.634Z [container-agent] 2024-05-08 18:19:49 INFO juju-log Updating Patroni config file
2024-05-08T18:19:49.981Z [container-agent] 2024-05-08 18:19:49 INFO juju-log Updated health checks
2024-05-08T18:20:50.758Z [container-agent] 2024-05-08 18:20:50 INFO juju-log Kubernetes service 'postgresql-k8s' patched successfully
2024-05-08T18:20:51.429Z [container-agent] 2024-05-08 18:20:51 INFO juju.worker.uniter.operation runhook.go:186 ran "update-status" hook (via hook dispatching script: dispatch)
2024-05-08T18:23:54.823Z [container-agent] 2024-05-08 18:23:54 INFO juju-log Updating Patroni config file
2024-05-08T18:23:55.142Z [container-agent] 2024-05-08 18:23:55 INFO juju-log Updated health checks
2024-05-08T18:24:55.901Z [container-agent] 2024-05-08 18:24:55 INFO juju-log Kubernetes service 'postgresql-k8s' patched successfully
2024-05-08T18:24:56.542Z [container-agent] 2024-05-08 18:24:56 INFO juju.worker.uniter.operation runhook.go:186 ran "update-status" hook (via hook dispatching script: dispatch)
2024-05-08T18:29:08.724Z [container-agent] 2024-05-08 18:29:08 INFO juju-log Updating Patroni config file
2024-05-08T18:29:09.052Z [container-agent] 2024-05-08 18:29:09 INFO juju-log Updated health checks
2024-05-08T18:30:09.787Z [container-agent] 2024-05-08 18:30:09 INFO juju-log Kubernetes service 'postgresql-k8s' patched successfully
2024-05-08T18:30:10.360Z [container-agent] 2024-05-08 18:30:10 INFO juju.worker.uniter.operation runhook.go:186 ran "update-status" hook (via hook dispatching script: dispatch)
2024-05-08T18:33:42.602Z [container-agent] 2024-05-08 18:33:42 INFO juju-log Updating Patroni config file
2024-05-08T18:33:42.931Z [container-agent] 2024-05-08 18:33:42 INFO juju-log Updated health checks
2024-05-08T18:34:43.772Z [container-agent] 2024-05-08 18:34:43 INFO juju-log Kubernetes service 'postgresql-k8s' patched successfully
2024-05-08T18:34:44.393Z [container-agent] 2024-05-08 18:34:44 INFO juju.worker.uniter.operation runhook.go:186 ran "update-status" hook (via hook dispatching script: dispatch)
2024-05-08T18:38:41.110Z [container-agent] 2024-05-08 18:38:41 INFO juju-log Updating Patroni config file
2024-05-08T18:39:41.810Z [container-agent] 2024-05-08 18:39:41 INFO juju-log Updated health checks
2024-05-08T18:40:42.681Z [container-agent] 2024-05-08 18:40:42 INFO juju-log Kubernetes service 'postgresql-k8s' patched successfully
2024-05-08T18:40:43.402Z [container-agent] 2024-05-08 18:40:43 INFO juju.worker.uniter.operation runhook.go:186 ran "update-status" hook (via hook dispatching script: dispatch)
2024-05-08T18:42:54.179Z [container-agent] 2024-05-08 18:42:54 INFO juju-log Updating Patroni config file
2024-05-08T18:42:54.547Z [container-agent] 2024-05-08 18:42:54 INFO juju-log Updated health checks
2024-05-08T18:43:55.337Z [container-agent] 2024-05-08 18:43:55 INFO juju-log Kubernetes service 'postgresql-k8s' patched successfully
2024-05-08T18:43:55.982Z [container-agent] 2024-05-08 18:43:55 INFO juju.worker.uniter.operation runhook.go:186 ran "update-status" hook (via hook dispatching script: dispatch)
2024-05-08T18:47:53.111Z [container-agent] 2024-05-08 18:47:53 INFO juju-log Updating Patroni config file
2024-05-08T18:48:53.802Z [container-agent] 2024-05-08 18:48:53 INFO juju-log Updated health checks
2024-05-08T18:49:54.546Z [container-agent] 2024-05-08 18:49:54 INFO juju-log Kubernetes service 'postgresql-k8s' patched successfully
2024-05-08T18:49:55.197Z [container-agent] 2024-05-08 18:49:55 INFO juju.worker.uniter.operation runhook.go:186 ran "update-status" hook (via hook dispatching script: dispatch)
2024-05-08T18:53:37.537Z [container-agent] 2024-05-08 18:53:37 INFO juju-log Updating Patroni config file
2024-05-08T18:53:37.889Z [container-agent] 2024-05-08 18:53:37 INFO juju-log Updated health checks
2024-05-08T18:54:38.767Z [container-agent] 2024-05-08 18:54:38 INFO juju-log Kubernetes service 'postgresql-k8s' patched successfully
2024-05-08T18:54:39.416Z [container-agent] 2024-05-08 18:54:39 INFO juju.worker.uniter.operation runhook.go:186 ran "update-status" hook (via hook dispatching script: dispatch)
2024-05-08T18:59:18.498Z [container-agent] 2024-05-08 18:59:18 INFO juju-log Updating Patroni config file
2024-05-08T18:59:18.879Z [container-agent] 2024-05-08 18:59:18 INFO juju-log Updated health checks
2024-05-08T19:00:19.652Z [container-agent] 2024-05-08 19:00:19 INFO juju-log Kubernetes service 'postgresql-k8s' patched successfully
2024-05-08T19:00:20.327Z [container-agent] 2024-05-08 19:00:20 INFO juju.worker.uniter.operation runhook.go:186 ran "update-status" hook (via hook dispatching script: dispatch)
2024-05-08T19:03:34.520Z [container-agent] 2024-05-08 19:03:34 INFO juju-log Updating Patroni config file
2024-05-08T19:03:34.865Z [container-agent] 2024-05-08 19:03:34 INFO juju-log Updated health checks
2024-05-08T19:04:35.609Z [container-agent] 2024-05-08 19:04:35 INFO juju-log Kubernetes service 'postgresql-k8s' patched successfully
2024-05-08T19:04:36.238Z [container-agent] 2024-05-08 19:04:36 INFO juju.worker.uniter.operation runhook.go:186 ran "update-status" hook (via hook dispatching script: dispatch)
2024-05-08T19:07:38.638Z [container-agent] 2024-05-08 19:07:38 INFO juju-log Updating Patroni config file
2024-05-08T19:07:38.957Z [container-agent] 2024-05-08 19:07:38 INFO juju-log Updated health checks
2024-05-08T19:08:39.749Z [container-agent] 2024-05-08 19:08:39 INFO juju-log Kubernetes service 'postgresql-k8s' patched successfully
2024-05-08T19:08:40.399Z [container-agent] 2024-05-08 19:08:40 INFO juju.worker.uniter.operation runhook.go:186 ran "update-status" hook (via hook dispatching script: dispatch)
2024-05-08T19:12:38.678Z [container-agent] 2024-05-08 19:12:38 INFO juju-log Updating Patroni config file
2024-05-08T19:12:39.010Z [container-agent] 2024-05-08 19:12:39 INFO juju-log Updated health checks
```

**Comment by mcfly722** [comment]
I tried for several times to redeploy whole model and juju controller, and found that this problem are stable occurs on current juju 3.5 version.
After I have downgraded juju version to 3.4.2/stable, there starts everything fine from first time.

**Comment by dragomirp** [comment]
Hi, @mcfly722, @Zvirovyi, sorry for the late response. There is an incompatibility between the postgresql-k8s charm and Juju 3.5 at the moment. Please, use 3.4 for the time being.

**Comment by marceloneppel** [comment]
Hi, @mcfly722 and @Zvirovyi!
Juju 3.5.1 is available in the `3.5/stable` snap channel, and PostgreSQL works with that.
